File: Week 2 Arrays/readability/readability.c
#include "cs50.h"
#include "math.h"
#include "ctype.h"
#include "stdio.h"
#include "string.h"

int calculate_level(string text);

int main()
{
  string text = get_string("Text: ");
  int count = calculate_level(text);

  // print the value
  if (count <= 1)
  {
    printf("Before Grade 1\n");
  }
  if (count >= 16)
  {
    printf("Grade 16+\n");
  }
  if (count > 1 && count < 16)
  {
    printf("Grade %i\n", count);
  }
}

int calculate_level(string text)
{
  char upperCase;
  int countLetters = 0;
  int countWords = 1;
  int countSentences = 0;

  // calculate letters, words and sentences
  for (int i = 0; i < strlen(text); i++)
  {
    upperCase = toupper(text[i]);
    if (upperCase >= 65 && upperCase <= 90)
    {
      countLetters++;
    }
    if (upperCase == ' ')
    {
      countWords++;
    }
    if (upperCase == '!' || upperCase == '.' || upperCase == '?')
    {
      countSentences++;
    }
  }

  // calculate and round the index value
  double L = (float)countLetters / countWords * 100;
  double S = (float)countSentences / countWords * 100;
  double index = round((float)0.0588 * L - 0.296 * S - 15.8);

  return index;
}

File: Week 6 Python/sentimental-readability/readability.py
# TODO

def main():
    text = input("Text: ")
    count = calculate_level(text)

    if count <= 1:
        print("Before Grade 1")
    if count >= 16:
        print("Grade 16+")
    if count > 1 and count < 16:
        print(f"Grade, {count}")


def calculate_level(text):
    upperCase = None
    countLetters = 0
    countWords = 1
    countSentences = 0

    # calculate letters, words and sentences
    for i in range(len(text)):
        # convert to uppercase
        upperCase = text[i].upper()
        # check if it's a letter, blank or punctuation
        # if so increase count
        if upperCase >= chr(65) and upperCase <= chr(90):
            countLetters += 1
        if upperCase == " ":
            countWords += 1
        if upperCase == '!' or upperCase == '.' or upperCase == '?':
            countSentences += 1

     # calculate and round the index value
    l = float(countLetters / countWords * 100)
    s = float(countSentences / countWords * 100)
    index = round(float(0.0588 * l - 0.296 * s - 15.8))

    return index


main()
